src/main/scala/com/snowplowanalytics/indicative/LambdaHandler.scala

```diff
@@ -12,33 +12,37 @@
  */
 package com.snowplowanalytics.indicative
 
-// Scala
 import scala.collection.JavaConverters._
 
-// cats
 import cats.data.EitherT
-import cats.effect.IO
+import cats.effect.{Clock, IO}
 import cats.implicits._
-
-// AWS
 import com.amazonaws.services.lambda.runtime.events.KinesisEvent
-
-// circe
-import io.circe.JsonObject
-
-// This library
-import com.snowplowanalytics.indicative.Transformer.TransformationError
 import com.snowplowanalytics.snowplow.analytics.scalasdk.json.EventTransformer
+import io.circe.Json
+
+import Transformer.TransformationError
 
 class LambdaHandler {
+  import LambdaHandler._
 
-  val apiKey: String =
-    sys.env.getOrElse("INDICATIVE_API_KEY",
-                      throw new RuntimeException("You must provide environment variable INDICATIVE_API_KEY"))
+  val apiKey: String = getConfig[String]("INDICATIVE_API_KEY")(s => s)
+  val unusedEvents
+    : List[String] = getConfig[List[String]]("UNUSED_EVENTS", Some(Filters.unusedEvents))(strToList(_)) // eg, `UNUSED_EVENTS=page_ping,app_heartbeat`
+  val unusedAtomicFields: List[String] = getConfig[List[String]](
+    "UNUSED_ATOMIC_FIELDS",
+    Some(Filters.unusedAtomicFields))(strToList(_)) // eg, `UNUSED_ATOMIC_FIELDS=etl_tstamp,geo_longitude`
+  val unusedContexts: List[String] = getConfig[List[String]]("UNUSED_CONTEXTS", Some(Filters.unusedContexts))(
+    strToList(_)) // eg, `UNUSED_CONTEXTS=performance_timing,geolocation_context`
+
+  // Number of events in a payload sent to Indicative is limited to 100
   val indicativeBatchSize = 100
+  // Size of a payload sent to Indicative is limited to 1Mb
+  val indicativePayloadBytesSize = 1000000
+  implicit val c: Clock[IO]      = Clock.create[IO]
 
   def recordHandler(event: KinesisEvent): Unit = {
-    val events: List[Either[TransformationError, JsonObject]] = event.getRecords.asScala
+    val events: List[Either[TransformationError, Json]] = event.getRecords.asScala
```

Comment:
I'd change to `...asScala.toList.flatMap(...)`

I also think that content of this `map` is a good candidate for a separate function.